Automattic/vip-cli
# [dev-env] Don't try to pull Dev Env images if connectivity is missing or last pull happened less than a week ago

## Description

We were forcing image pull every time on `start`, in situations where ghcr.io (where we store images) is not resolvable it led to start command dying, even though the images could be downloaded and available locally.

This PR addresses this behavior by only forcing pull if the previous pull was more than a week ago and `ghcr.io` is resolvable. 

We're using `pre-engine-build` hook to force Lando to thing the defined services are local, rather than pullable. 

In practice this means that if an image was recently downloaded we won't try to download it. If an image is missing it would try to pull. If a registry is resolvable this "just works", if it's not resolvable it'll die with an error on the image that's missing. 

## Steps to Test

1. Check out the PR, build and `npm link`
2. Create an env
3. Start it, wait for it to be fully set up
4. Disconnect from the internet
5. Restart the environment and verify that it starts just fine even without the connection.

## Props

@sjinks for proposing a simpler solution not requiring a Lando fork patch.
